modules/shadertools/src/modules/lights/lights.js

```diff
@@ -46,6 +46,22 @@ function getLightSourceUniforms({ambientLight, pointLights = [], directionalLigh
 }
 
 function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
+  // Support for array of lights. Type of light is detected by field
+  // TODO - this should work directly against the uniforms, if we phase out `opts.lightSources`
+  if ('lights' in opts) {
+    const lightSources = {pointLights: [], directionalLights: []};
+    for (const light of opts.lights || []) {
+      if (light.position) {
```

Comment:
this is not safe if we want to make light to have generic props in the short future. We can say it is a point light when having a position prop, same as directional light.

Reply:
Yes I see your point. I do find it inefficient that the user has to "sort" the light objects into different arrays. 

I think that the user should just be able to put any classes inheriting from the base `Light` in the same array.

I think it should still be possible to make it equally extensible. Happy to implement it differently if you agree with that idea.

Reply:
No problem, got your idea. Just implement it differently. Thanks!